File: final_project/frontend/src/models/RelayStory.js
/**
 * RelayStory 모델 클래스
 * 릴레이 스토리의 다음 이야기(댓글) 정보를 관리하는 클래스
 */
export class RelayStory {
  constructor({
    id = null,
    content = '',
    author = '',
    userId = null,
    postId = null,
    createdAt = null,
    updatedAt = null
  } = {}) {
    this.id = id
    this.content = content
    this.author = author
    this.userId = userId
    this.postId = postId
    this.createdAt = createdAt
    this.updatedAt = updatedAt
  }

  /**
   * 서버 응답을 RelayStory 객체로 변환
   */
  static fromJSON(data) {
    return new RelayStory({
      id: data.id,
      content: data.content,
      author: data.author,
      userId: data.userId,
      postId: data.postId,
      createdAt: data.createdAt,
      updatedAt: data.updatedAt
    })
  }

  /**
   * RelayStory 객체를 JSON으로 변환 (API 전송용)
   */
  toJSON() {
    return {
      id: this.id,
      content: this.content,
      author: this.author,
      userId: this.userId,
      postId: this.postId,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt
    }
  }

  /**
   * 다음 이야기가 유효한지 확인
   */
  isValid() {
    return !!(this.content && this.postId)
  }

  /**
   * 다음 이야기가 비어있는지 확인
   */
  isEmpty() {
    return !this.content || this.content.trim() === ''
  }

  /**
   * 내용 길이 반환
   */
  getContentLength() {
    return this.content.length
  }
}

File: final_project/backend/models/relay_story.py
"""
RelayStory Pydantic models for API validation and serialization
"""

from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional


class RelayStoryCreate(BaseModel):
  """Model for creating a new relay story (comment)"""
  content: str = Field(..., min_length=1)
  author: str = Field(..., min_length=1, max_length=50)
  post_id: int
  user_id: int


class RelayStoryUpdate(BaseModel):
  """Model for updating a relay story"""
  content: Optional[str] = Field(None, min_length=1)
  author: Optional[str] = Field(None, min_length=1, max_length=50)


class RelayStory(BaseModel):
  """RelayStory model for database"""
  id: Optional[int] = None
  content: str = Field(..., min_length=1)
  author: str = Field(..., min_length=1, max_length=50)
  user_id: int
  post_id: int
  created_at: datetime = Field(default_factory=datetime.utcnow)
  updated_at: datetime = Field(default_factory=datetime.utcnow)

  class Config:
      from_attributes = True


class RelayStoryResponse(BaseModel):
  """Response model for relay story information"""
  id: int
  content: str
  author: str
  user_id: int
  post_id: int
  created_at: datetime
  updated_at: datetime

  class Config:
    from_attributes = True
